src/main.ts

```diff
@@ -62,11 +84,40 @@ export async function runSorald(
   return allRepairs;
 }
 
+/**
+ * Filter out any violation specifiers that aren't present in the changed lines
+ * of code.
+ */
+function filterViolationSpecsByRatchet(
+  violationSpecs: string[],
+  changedLines: Map<PathLike, ClosedRange[]> | undefined,
+  source: PathLike
+): string[] {
+  if (changedLines === undefined) {
+    return violationSpecs;
+  }
+
+  return violationSpecs.filter(spec => {
+    const filePath = path.join(source.toString(), sorald.parseFilePath(spec));
+    const lineRange = sorald.parseAffectedLines(spec);
+    const changedLinesInFile = changedLines.get(filePath);
+    return (
+      changedLinesInFile !== undefined &&
+      ranges.overlapsAny(lineRange, changedLinesInFile)
+    );
+  });
+}
+
 async function run(): Promise<void> {
   try {
     const source: PathLike = core.getInput('source');
     const soraldJarUrl: string = core.getInput('sorald-jar-url');
-    const repairedViolations: string[] = await runSorald(source, soraldJarUrl);
+    const ratchetFrom: string | undefined = core.getInput('ratchet-from');
+    const repairedViolations: string[] = await runSorald(
+      source,
+      soraldJarUrl,
+      ratchetFrom ? ratchetFrom : undefined
```

Comment:
Is this ternary operator required? Line 115 could be a string or undefined so accordingly the value can be set here. However, if the string is `''` (empty string), you may want to set it to `undefined`.

Reply:
Actually, line 115 is lying, it can't be `undefined`. It's always a string (empty string if it's not defined). So I need to change that confusing type annotation.